File: 백준/Bronze/27930. 당신은 운명을 믿나요？/당신은 운명을 믿나요？.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split("")
// let input = fs.readFileSync('./EX.txt').toString().trim().split("")

const korea = "KOREA"
let kIdx = 0
const yonsei = "YONSEI"
let yIdx = 0

while (input.length) {
    let x = input.shift()

    if (x === korea[kIdx]) {{
        kIdx += 1
    }}
    if (x === yonsei[yIdx]) {
        yIdx += 1
    }

    if (kIdx === 5){
        answer = korea
        break
    } else if (yIdx === 6) {
        answer = yonsei
        break
    }
}
console.log(answer)

File: 백준/Bronze/27930. 당신은 운명을 믿나요？/당신은 운명을 믿나요？.py
import sys
input = sys.stdin.readline

S = list(i for i in input().rstrip())

korea = "KOREA"
yonsei = "YONSEI"
k_idx = 0
y_idx = 0

while S:
    x = S.pop(0)

    if korea[k_idx] == x:
        k_idx += 1
    if yonsei[y_idx] == x:
        y_idx += 1


    if k_idx == 5:
        print(korea)
        break
    if y_idx == 6:
        print(yonsei)
        break

